app/src/main/java/com/geeksville/mesh/ui/MapFragment.kt

```diff
@@ -1,680 +1,316 @@
-     */
-    private val longClick = OnMapLongClickListener {
-        val userDefinedPointImg =
-            ContextCompat.getDrawable(
-                requireActivity(),
-                R.drawable.baseline_location_on_white_24dp
-            )!!
-                .toBitmap()
-        point = Point.fromLngLat(it.longitude(), it.latitude())
-
-
-        /*
-        Calculate region from user specified position.
-        5 miles NE,NW,SE,SW from user center point.
-        25 Sq Mile Region
-        */
-        //____________________________________________________________________________________________
-        val topRight = calculateCoordinate(45.0, point?.latitude()!!, point?.longitude()!!)
-        val topLeft = calculateCoordinate(135.0, point?.latitude()!!, point?.longitude()!!)
-        val bottomLeft = calculateCoordinate(225.0, point?.latitude()!!, point?.longitude()!!)
-        val bottomRight = calculateCoordinate(315.0, point?.latitude()!!, point?.longitude()!!)
-        //____________________________________________________________________________________________
-
-        val pointList = listOf(topRight, topLeft, bottomLeft, bottomRight, topRight)
-
-        squareRegion = LineString.fromLngLats(pointList)
-
-        geoJsonSource = geoJsonSource(boundingBoxId) {
-            geometry(squareRegion)
-        }
-        lineLayer = lineLayer(lineLayerId, boundingBoxId) {
-            lineCap(LineCap.ROUND)
-            lineJoin(LineJoin.MITER)
-            lineOpacity(0.7)
-            lineWidth(1.5)
-            lineColor("#888")
-        }
-
-        if (point != null) {
-            binding.downloadRegion.visibility = View.VISIBLE
+    override fun onResume() {
+        super.onResume()
+        map.invalidate()
+        val tileSourceName = mPrefs.getString(
+            prefsTileSource,
+            TileSourceFactory.DEFAULT_TILE_SOURCE.name()
+        )
+        try {
+            map.setTileSource(matchOnlineTileSourceBase(tileSourceName!!))
+        } catch (e: IllegalArgumentException) {
+            map.setTileSource(TileSourceFactory.DEFAULT_TILE_SOURCE)
         }
+        map.onResume()
+    }
 
-        mapView?.getMapboxMap()?.getStyle()?.let { style ->
-            userTouchPosition.geometry(point!!)
-            if (!style.styleLayerExists(userTouchLayerId)) {
-                style.addImage(userPointImageId, userDefinedPointImg)
-                style.addSource(userTouchPosition)
-                style.addSource(geoJsonSource)
-                style.addPersistentLayer(lineLayer)
-                style.addLayer(userTouchLayer)
-            } else {
-                style.removeStyleLayer(lineLayerId)
-                style.removeStyleSource(boundingBoxId)
-                style.addSource(geoJsonSource)
-                style.addLayer(lineLayer)
-            }
+    private fun matchOnlineTileSourceBase(name: String): OnlineTileSourceBase {
```

Comment:
I would like to try and make this more dynamic, Otherwise every time we add a new source it must be added here

Reply:
Not needed after removing onPause() functionality